chain/reorg.go

```diff
@@ -283,44 +414,83 @@ func (reorg *reorganizer) gatherChainInfo() error {
 	return ErrNotExistBranchRoot
 }
 
-func (reorg *reorganizer) rollbackChain() error {
+// build reorg chain info from marker
+func (reorg *reorganizer) gatherReco() error {
+	var err error
+
+	cdb := reorg.cs.cdb
+
+	startBlock := reorg.brStartBlock
+	bestBlock := reorg.bestBlock
+	topBlock := reorg.brTopBlock
+
+	reorg.brStartBlock = startBlock
+	reorg.bestBlock = bestBlock
+
+	for tmpBlk := bestBlock; tmpBlk.GetHeader().GetBlockNo() > startBlock.GetHeader().GetBlockNo(); {
```

Comment:
The two loops in this method are almost the same. The duplicates would rather be removed.

Reply:
I refactored that part.